src/com/sap/piper/tools/ToolType.groovy

```diff
@@ -0,0 +1,38 @@
+package com.sap.piper.tools
+
+
+enum ToolType {
+
+    JAVA ('Java', ToolKey.JAVA, ToolExecutable.JAVA, ToolVersion.JAVA), 
+    MTA ('SAP Multitarget Application Archive Builder', ToolKey.MTA, ToolExecutable.MTA, ToolVersion.MTA),
```

Comment:
No, I don't think it would be better.
No, so far we don't expect it to be reused by multiple steps, but we expect to reuse the code of ToolUtils and ToolVerifier by multiple tools.
Yes, in case that there is new steps with new tools, new tool types should be added.